### ISSUE TITLE
Output key id as a property of the retrieved queue message

### ISSUE THREAD

**Issue by draggeta** [Azure/azure-storage-python#528]
### Which service(blob, file, queue) does this issue concern?
Queue

### Which version of the SDK was used? Please provide the output of `pip freeze`.
1.3.0

### What problem was encountered?
I'm using a keyResolver to decrypt the message. This works and I don't have any issues with that. I would like however, to have an option to also output the Key ID used to decrypt the message. For us, the key ID is the frontend it came from and if a message fails to decrypt, this would be the easiest solution to debug this.

### Have you found a mitigation/solution?
No easy ones. You can work around it by having the frontends write to a table which jobs they posted, but it is not as clean.

Note: for table service, please post the issue here instead: https://github.com/Azure/azure-cosmosdb-python.

**Comment by rickle-msft** [comment]
Hi, @draggeta. Thank you for posting your question. I have logged this for discussion with our team, and we will discuss the possibility of adding this feature. Perhaps we can find a workaround for you in the meantime, though. Encrypted queue messages are actually stored as a json dictionary of encrypted data with encryption metadata. It should be possible for you to set a response callback on the queueservice that inspects the still encrypted message and extracts the keyID from this metadata. You can find the format of the metadata [here](https://github.com/Azure/azure-storage-python/blob/master/azure-storage-common/azure/storage/common/_encryption.py#L111) and the structure of the encrypted queue message [here](https://github.com/Azure/azure-storage-python/blob/master/azure-storage-queue/azure/storage/queue/_encryption.py#L80). Let me know if you have trouble getting the info you want out of the response.

Can you please remind me what message you currently get when decryption fails?

**Comment by draggeta** [comment]
Hi @rickle-msft, I'm not sure what you mean by a response callback, but I'll try and look into it.

Either way, the error I get is:

```
Client-Request-ID=11110000-1111-0000-1111-000011110000 Encountered decryption failure: this cannot be retried. Server-Timestamp=Mon, 19 Nov 2018 20:47:31 GMT, Server-Request-ID=11110000-1111-0000-1111-000011110000, HTTP status code=200, Exception=Decryption failed.
```

**Comment by rickle-msft** [comment]
Sorry about that. You can take a look [here](https://github.com/Azure/azure-storage-python/blob/master/azure-storage-common/azure/storage/common/storageclient.py#L110) for more information on the response callback. Essentially you can set a function on the queueservice object that will be called as soon as we get the response back so you can do anything you want with the raw response. You should be able to grab the keyID before we decrypt the message and discard the encryption info.

What do you mean when you say pass it a value? What value are you passing in what context? And have you been able to decrypt at all in any cases?

**Comment by draggeta** [comment]
Ah, thank you. After reading about it, it was clear how I should be using it conceptually, but couldn't figure out where.

As for your question. I was very unclear there. Should do less late night typing. I meant that you never see the encrypted message. Hopefully I'll get it to work. If so, then the issue is a bit moot.

**Comment by rickle-msft** [comment]
Sounds good. Let me know if you need any more guidance

**Comment by draggeta** [comment]
Well, I can get it to work and for now the loggin functionality works. Sadly, you cannot return any data, so anything I need to perform on invalid/incorrect messages, will need to happen in the callback. 

Having the requested functionality would still be nice IMHO, but for now it is more of a nice to have instead of a need.

**Comment by rickle-msft** [comment]
I'm glad you're unblocked for now! We haven't yet had a team meeting on feature requests since you posted this, so the discussion is still pending, but one of us will update you whenever we make a decision about whether/when we will add this feature. Thank you again for your feedback!
